components/tidb_query/src/rpn_expr/impl_arithmetic.rs

```diff
@@ -1220,17 +1279,17 @@ mod tests {
             ),
         ];
 
-        for (lhs, rhs) in abnormal {
+        for args in cases {
             for &(flag, sql_mode, is_ok, has_warning) in &modes {
                 // Construct an `EvalContext`
                 let mut config = EvalConfig::new();
                 config.set_flag(flag).set_sql_mode(sql_mode);
 
-                let (result, mut ctx) = RpnFnScalarEvaluator::new()
+                let evaluator_builder = RpnFnScalarEvaluator::new()
```

Comment:
why not `RpnFnScalarEvaluator::new().context(..).push_params(..).evaluate_ctx::<T>(..)`?